samples/apply-dictionary-renderer-to-graphics-overlay/src/main/java/com/esri/arcgismaps/sample/applydictionaryrenderertographicsoverlay/components/ApplyDictionaryRendererToGraphicsOverlayViewModel.kt

```diff
@@ -0,0 +1,252 @@
+/* Copyright 2025 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.applydictionaryrenderertographicsoverlay.components
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.viewModelScope
+import com.arcgismaps.geometry.Multipoint
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.geometry.SpatialReference
+import com.arcgismaps.mapping.ArcGISScene
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.PortalItem
+import com.arcgismaps.mapping.symbology.DictionaryRenderer
+import com.arcgismaps.mapping.symbology.DictionarySymbolStyle
+import com.arcgismaps.mapping.view.Camera
+import com.arcgismaps.mapping.view.Graphic
+import com.arcgismaps.mapping.view.GraphicsOverlay
+import com.arcgismaps.portal.Portal
+import com.arcgismaps.toolkit.geoviewcompose.SceneViewProxy
+import com.esri.arcgismaps.sample.applydictionaryrenderertographicsoverlay.R
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialogViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import org.xmlpull.v1.XmlPullParser
+import org.xmlpull.v1.XmlPullParserFactory
+import java.io.File
+
+/**
+ * ViewModel for the "Apply dictionary renderer to graphics overlay" sample.
+ */
+class ApplyDictionaryRendererToGraphicsOverlayViewModel(private val app: Application) : AndroidViewModel(app) {
+
+    // The scene shown in the SceneView composable
+    val arcGISScene = ArcGISScene(basemapStyle = BasemapStyle.ArcGISTopographic)
+
+    // Graphics overlay that will hold the message graphics
+    val graphicsOverlay: GraphicsOverlay = GraphicsOverlay()
+
+    // SceneViewProxy to enable programmatic viewpoint changes
+    val sceneViewProxy = SceneViewProxy()
+
+    // Used to display error messages
+    val messageDialogVM = MessageDialogViewModel()
+
+    // Provision path where downloaded sample assets will be placed by the downloader activity
+    private val provisionPath: String by lazy {
+        app.getExternalFilesDir(null)?.path.toString() + File.separator + app.getString(R.string.apply_dictionary_renderer_to_graphics_overlay_app_name)
+    }
+
+    init {
+        viewModelScope.launch(Dispatchers.IO) {
```

Comment:
The local file should be provisioned prior to the init, probably could remove `.IO`, or let me know if its needed: 
```suggestion
        viewModelScope.launch {
```